packages/http-client-python/generator/pygen/codegen/models/client.py

```diff
@@ -79,6 +79,26 @@ def __init__(
         self.request_id_header_name = self.yaml_data.get("requestIdHeaderName", None)
         self.has_etag: bool = yaml_data.get("hasEtag", False)
 
+        # update the host parameter value. In later logic, SDK will overwrite it
+        # with value from cloud_setting if users don't provide it.
+        if self.need_cloud_setting:
+            for p in self.parameters.parameters:
+                if p.location == ParameterLocation.ENDPOINT_PATH:
+                    p.client_default_value = ""
```

Comment:
can we use a `_DEFAULT` instead?

Reply:
See comment here https://github.com/Azure/autorest.python/pull/3041#discussion_r2033617927

Reply:
with offline discussion, i think it's better to make this parameter to be optional.

Reply:
ok i'm good with defaulting to `None` since that will never be a valid endpoint value